Tasks/VsTest/distributedTest.ts

```diff
@@ -0,0 +1,190 @@
+import * as tl from 'vsts-task-lib/task';
+import * as webapim from './node_modules/vso-node-api/WebApi';
+import * as testInterfaces from './node_modules/vso-node-api/interfaces/TestInterfaces';
+import * as testApis from './node_modules/vso-node-api/TestApi';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as tr from 'vsts-task-lib/toolrunner';
+import * as models from './models';
+import * as Q from 'q';
+import * as ps from 'child_process';
+
+export class DistributedTest {
+
+    constructor(dtaTestConfig: models.dtaTestConfigurations) {
+        this.dtaPid = -1;
+        this.dtaTestConfig = dtaTestConfig;
+        this.tfsCollectionUrl = tl.getVariable('System.TeamFoundationCollectionUri');
+        this.patToken = tl.getEndpointAuthorization('SystemVssConnection', true).parameters['AccessToken'];
+        const releaseId = tl.getVariable('Release.ReleaseId');
+        const phaseId = tl.getVariable('Release.DeployPhaseId');
+        const projectName = tl.getVariable('System.TeamProject');
+        const taskInstanceId = this.getDtaInstanceId();
+        this.environmentUri = 'dta://env/' + projectName + '/_apis/release/' + releaseId + '/' + phaseId + '/' + taskInstanceId;
+        this.testAgentName = tl.getVariable('Agent.MachineName');
+    }
+
+    public runDistributedTest() {
+        this.registerAndConfigureAgent();
+    }
+
+    private registerAndConfigureAgent() {
+        const webapi = this.getWebApi();
+        const testApi = webapi.getTestApi();
+        const envUrlRef: any = { Url: this.environmentUri };
+        const machineNameRef = { Name: this.testAgentName };
+        const testAgent =
```

Comment:
>testAgent [](start = 14, length = 9)

from a code structure perspective.. testagent could be a separate class.. can take it up later.. register etc.. are methods in it.